Sources/Datadog/Core/Upload/DataUploader.swift

```diff
@@ -9,21 +9,28 @@ import Foundation
 /// Creates URL and adds query items before providing them
 internal class UploadURLProvider {
     private let urlWithClientToken: URL
-    private let dateProvider: DateProvider
+    private let queryItems: [QueryItem]
 
-    private var queryItems: [URLQueryItem] {
-        // batch_time
-        let timestamp = dateProvider.currentDate().timeIntervalSince1970.toMilliseconds
-        let batchTimeQueryItem = URLQueryItem(name: "batch_time", value: "\(timestamp)")
-        // ddsource
-        let ddSourceQueryItem = URLQueryItem(name: "ddsource", value: "ios")
+    struct QueryItem {
+        let value: () -> URLQueryItem
```

Comment:
maybe that doesn't need to be a new type
```swift
typealias QueryItemProvider = () -> URLQueryItem
extension URLQueryItem {
  static func batchTimeProvider() -> QueryItemProvider {
    return { URLQueryItem(...) }
  }
}
```

if you don't like the fact that we can't extend this `typealias`, then we should at least make `QueryItemProvider` a `class` otherwise its `value` get copied every time it's passed

Reply:
Yay 🙌, I did `QueryItemProvider`, but I kept the `.batchTime(...)` as this would be too provider-ish (`UploadURLProvider.QueryItemProvider.batchTimeProvider()` 🙅).

Reply:
👍 can we also make it a `class` please?

Reply:
👍 it's a `class` now.

Reply:
_for the records:_ changing to `class` seems to bring a tiny performance gain, which is inconsistent with my comment. it's still good to keep it as `class` as it doesn't have any downsides neither